### PR TITLE
Move js-specific URL parser tests into their own file

### LINKED ISSUE #46941: Consider removing url test containing utf16 surrogates
`urltestdata.json` contains the following test:
```json
{
    "input": "http://example.com/\uD800\uD801\uDFFE\uDFFF\uFDD0\uFDCF\uFDEF\uFDF0\uFFFE\uFFFF?\uD800\uD801\uDFFE\uDFFF\uFDD0\uFDCF\uFDEF\uFDF0\uFFFE\uFFFF",
    "base": null,
    "href": "http://example.com/%EF%BF%BD%F0%90%9F%BE%EF%BF%BD%EF%B7%90%EF%B7%8F%EF%B7%AF%EF%B7%B0%EF%BF%BE%EF%BF%BF?%EF%BF%BD%F0%90%9F%BE%EF%BF%BD%EF%B7%90%EF%B7%8F%EF%B7%AF%EF%B7%B0%EF%BF%BE%EF%BF%BF",
    "origin": "http://example.com",
    "protocol": "http:",
    "username": "",
    "password": "",
    "host": "example.com",
    "hostname": "example.com",
    "port": "",
    "pathname": "/%EF%BF%BD%F0%90%9F%BE%EF%BF%BD%EF%B7%90%EF%B7%8F%EF%B7%AF%EF%B7%B0%EF%BF%BE%EF%BF%BF",
    "search": "?%EF%BF%BD%F0%90%9F%BE%EF%BF%BD%EF%B7%90%EF%B7%8F%EF%B7%AF%EF%B7%B0%EF%BF%BE%EF%BF%BF",
    "hash": ""
  }
```
The interesting part here is the `\uD801\uDFFE`  - that's a UTF-16 surrogate.

The behaviour in this case is undefined as per the URL specification[^5], where the input to the url parsing algorithm is a scalar value string[^2] (meaning a string containing neither leading nor trailing surrogate characters).

`url/README.md` states:
> `resources/urltestdata.json` contains URL parsing tests suitable for any URL parser implementation.

Therefore, the suite should *only* test behaviour defined in the url specification[^4].

I would like to hear the thoughts of more qualified people on this before I make a PR for it (:

[^1]: https://262.ecma-international.org/6.0/#sec-ecmascript-language-types-string-type
[^2]: https://infra.spec.whatwg.org/#scalar-value-string
[^3]: https://github.com/json5/json5-spec/issues/12
[^4]: https://url.spec.whatwg.org
[^5]:https://url.spec.whatwg.org/#url-parsing

### PR DESCRIPTION
**Pull request by simonwuelker**
The test in question contains UTF-16 surrogates, but the input to a url parser is defined to be a "scalar value string", which is a string containing *no* surrogates.

Fixes https://github.com/web-platform-tests/wpt/issues/46941, refer to this issue for more information.

### CODE REVIEW

**Comment by annevk** [comment]
I think you are correct that if you feed this inputs directly into the parser it would be wrong, but we're feeding these inputs into `new URL()` which takes `USVString` which handles the conversion.

I think it's fair to say we should decouple that from this JSON file so it's useful more generally, but I think that would require adding this as a standalone test somehow that browsers and other JS clients (thinking of Deno and Node.js) would still run as it's very meaningful to them.

**Comment by domenic** [comment]
Concrete proposal: break urltestdata.json up into two files, maybe urltestdata.json and urltestdata-javascript-only.json. The latter would get a comment at the top explaining the very rare cases in which it should be used. And then modify all the tests that consume urltestdata.json to consume both files.

File: url/README.md [hunk 1]
```diff
@@ -1,11 +1,13 @@
-## urltestdata.json
+## urltestdata.json / urltestdata-javascript-only.json
 
 `resources/urltestdata.json` contains URL parsing tests suitable for any URL parser implementation.
+`resources/urltestdata-javascript-only.json` contains URL parsing tests specifically meant 
+for javascripts `new URL()` function.
```
Comment (annevk) [inline url/README.md:5]:
```suggestion
`resources/urltestdata-javascript-only.json` contains URL parsing tests specifically meant 
for JavaScript's `URL()` class as well as other languages accepting non-scalar-value strings.
```

File: url/resources/urltestdata-javascript-only.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        "username": "",
+        "password": "",
+        "host": "example.com",
+        "hostname": "example.com",
+        "port": "",
+        "pathname": "/%EF%BF%BD%F0%90%9F%BE%EF%BF%BD%EF%B7%90%EF%B7%8F%EF%B7%AF%EF%B7%B0%EF%BF%BE%EF%BF%BF",
+        "search": "?%EF%BF%BD%F0%90%9F%BE%EF%BF%BD%EF%B7%90%EF%B7%8F%EF%B7%AF%EF%B7%B0%EF%BF%BE%EF%BF%BF",
+        "hash": ""
+      }
+]
```
Comment (annevk) [inline url/resources/urltestdata-javascript-only.json:18]: Please add a newline.

**Review by annevk — COMMENTED** [review]
Thanks, couple nits from my side, but with that this seems reasonable.

@lucacasonato @jasnell any concerns from your side?

**Comment by annevk** [comment]
Also seems you have some trailing whitespace to remove as per the lint bot.

**Review by domenic — APPROVED** [review]

**Pull request merged**